reproschema/utils.py

```diff
@@ -52,3 +53,105 @@ def start_server(port=8000, path=None, tmpdir=None):
 def stop_server(stop):
     stop()
     requests_cache.clear()
+
+
+# items that have to be fixed in the old schema
+LANG_FIX = [
+    "http://schema.org/schemaVersion",
+    "http://schema.org/version",
+    "http://schema.repronim.org/limit",
+    "http://schema.repronim.org/randomMaxDelay",
+    "http://schema.org/inLanguage",
+    "http://schema.repronim.org/schedule",
+]
+BOOL_FIX = [
+    "http://schema.repronim.org/shuffle",
+    "http://schema.org/readonlyValue",
+    "http://schema.repronim.org/multipleChoice",
+    "http://schema.org/valueRequired",
+]
+
+ALLOWTYPE_FIX = ["http://schema.repronim.org/allow"]
+ALLOWTYPE_MAPPING = {
+    "http://schema.repronim.org/Skipped": "http://schema.repronim.org/AllowSkip",
+    "http://schema.repronim.org/DontKnow": "http://schema.repronim.org/AllowAltResponse",
+}
+
+IMAGE_FIX = ["http://schema.org/image"]
+
+
+def _lang_fix(data_el):
+    if isinstance(data_el, dict):
+        data_el.pop("@language", None)
+    elif isinstance(data_el, list) and len(data_el) == 1:
+        data_el = data_el[0]
+        data_el.pop("@language", None)
+    else:
+        raise Exception(f"expected a list or dictionary, got {data_el}")
+    return data_el
+
+
+def _image_fix(data_el):
+    if isinstance(data_el, dict):
+        if "@id" not in data_el:
+            data_el["@id"] = data_el.pop("@value")
+            data_el.pop("@language", None)
+    elif isinstance(data_el, list) and len(data_el) == 1:
+        data_el = data_el[0]
+        data_el = _image_fix(data_el)
+    else:
+        raise Exception(f"expected a list or dictionary, got {data_el}")
+    return data_el
+
+
+def _bool_fix(data_el):
+    if isinstance(data_el, dict):
+        data_el["@type"] = "http://www.w3.org/2001/XMLSchema#boolean"
+    elif isinstance(data_el, list):
+        for el in data_el:
+            _bool_fix(el)
+    else:
+        raise Exception(f"expected a list or dictionary, got {data_el}")
+
+
+def _allowtype_fix(data_el):
+    if isinstance(data_el, dict):
+        if data_el["@id"] in ALLOWTYPE_MAPPING:
+            data_el["@id"] = ALLOWTYPE_MAPPING[data_el["@id"]]
+    elif isinstance(data_el, list):
+        for el in data_el:
+            _allowtype_fix(el)
+    else:
+        raise Exception(f"expected a list or dictionary, got {data_el}")
+
+
+def fixing_old_schema(data, copy_data=False):
```

Comment:
should we call this `migrate_schema`, include a version check, and set `schemaVersion` to `1.0.0`?